craft_providers/actions/snap_installer.py

```diff
@@ -180,6 +180,94 @@ def _get_host_snap(snap_name: str) -> Iterator[pathlib.Path]:
         yield snap_path
 
 
+def _get_assertion(query: List[str]) -> bytes:
+    """Get an assertion from snapd.
+
+    :param query: assertion query to pass to `snap known`
+    :returns: assertion data
+    :raises SnapInstallationError: if 'snap known' call fails
+    """
+    command = snap_cmd.formulate_known_command(query=query)
+    logger.debug("Executing command on host: %s", shlex.join(command))
```

Comment:
Just log "command". Yes, it's ugly in the sense of "a list with strings", but when a behaviour is weird and you have to debug it, having *exactly* what you executed is better.